src/docker/shared.rs

```diff
@@ -499,8 +512,49 @@ pub(crate) fn docker_envvars(
     Ok(())
 }
 
-pub(crate) fn docker_cwd(docker: &mut Command, paths: &DockerPaths) -> Result<()> {
+fn mount_to_ignore_cargo_config(
+    docker: &mut Command,
+    paths: &DockerPaths,
+    ignore_cargo_config: bool,
+) -> Result<()> {
+    let check_mount =
+        |cmd: &mut Command, host: &Path, mount: &Path, relpath: &Path| -> Result<()> {
+            let cargo_dir = relpath.join(".cargo");
+            if host.join(&cargo_dir).exists() {
+                // this is fine, since it has to be a POSIX path on the mount.
+                cmd.args(&["-v", &mount.join(&cargo_dir).as_posix()?]);
+            }
+
+            Ok(())
+        };
+    let is_cargo_home_parent = paths
+        .cargo_home()
+        .parent()
+        .map(|x| x == paths.host_root())
+        .unwrap_or_default();
+    if ignore_cargo_config && !is_cargo_home_parent {
+        let mount_root = Path::new(paths.mount_root());
+        let mount_cwd = Path::new(paths.mount_cwd());
+        check_mount(docker, &paths.cwd, mount_cwd, Path::new(""))?;
+        // CWD isn't guaranteed to be a subdirectory of the mount root.
```

Comment:
Since we've removed `/project`, this changes the logic here, so we don't necessarily have a relpath to the workspace root.